### ISSUE TITLE
Minion mip robot

### ISSUE THREAD

**Issue by JoJoBabahut** [WowWeeLabs/MiP-Android-SDK#6]
Hi,

Is the mip-android-sdk compatible with the new fun minion mip robot ?
If not, could you please update the sdk to support this robot ?

thanks in advance

**Comment by sailimc** [comment]
Hello,

Working on the compatibility with Minion MiP.
Should be ready in a few weeks.

Cheers
Forrest

**Comment by otahboub** [comment]
Hello,
 Is there a time line on opening the SDK to enable multiple MiP connections ? The SDK so far only enables a single MiP Robot. I currently trying to use MiP robots for a research project and I am in need to be able to create an Android application that enables me to control multiple MiPs (as in the app found in the App Store).

Thanks,

Omar

**Comment by sailimc** [comment]
Hello Omar,

Are you able to get a list of MiPs from the function below?
`List<MipRobot> mipFoundList = MipRobotFinder.getInstance().getMipsFoundList();`

This function should return a list of MipRobots scanned. Then you should be able to call the `connect` function of each MipRobot object to connect to all of them.

Cheers
Forrest

**Comment by otahboub** [comment]
Hello Forrest,

Actually the function will always return 1 if one or more MiP robots exists (and switched on).
I did insert a break point on line 315 in Class MainMenuActivity function (BroadcastReceiver), the function returned a List of a single robot (although there were a 2nd MiP robot).

 I attempted to debug further into the MipRobotFinder and realized that in the class in method handleFoundBluetoothDevice line-275 an intent object ("com.wowwee.bluetoothrobotcontrollib.MipRobotFinder_MipFound"), then it is broadcasted to the MainMenuActivity object (mMipFinderBroadcastReceiver) object at 308, where the intent of a MiP device is found is intercepted, the condition at line 316 evaluates true since the MipRobotFinder has returned the list with 1 robot (with out inserting the second robot to the found list) then the rest of the function  connects to the first Mip robot found in the list.

In conclusion, is that the MipRobotFinder immediately broadcasts the MipFound intent before adding the rest of detected Mip robots to the FoundList. Since the code is part of the SDK, I am not able to modify it to work with my application.

I hope this helps :) ! 

Thanks,

Omar

**Comment by otahboub** [comment]
Hi Forrest,

 I followed the code snippet on the main page for connecting multiple MiP it is now working ! 

Thanks,

Omar.

**Comment by sailimc** [comment]
Hi Omar,

Glad it is working. I do understand the problem you had, yes the current implementation might be a little confusing. Thanks for pointing that out. We will surely look into making it more easy to use.

Cheers
Forrest
